cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -1005,4 +1013,57 @@ private ClassLoader setContextCombinedClassLoader(BasicMapReduceContext context)
     return ClassLoaders.setContextClassLoader(new CombineClassLoader(
       null, ImmutableList.of(context.getProgram().getClassLoader(), getClass().getClassLoader())));
   }
+
+  /**
+   * Localizes resources requested by users in the MapReduce Program's beforeSubmit phase.
+   * In Local mode, also copies resources to a temporary directory.
+   *
+   * @param job the {@link Job} for this MapReduce program
+   * @param targetDir in local mode, a temporary directory to copy the resources to
+   * @return a {@link Map} of resource name to the {@link File} handle for the local file.
+   */
+  private Map<String, File> localizeUserResources(Job job, File targetDir) throws IOException {
+    Map<String, File> localizedResources = new HashMap<>();
+    Map<String, LocalizeResource> resourcesToLocalize = context.getResourcesToLocalize();
+    for (final Map.Entry<String, LocalizeResource> entry : resourcesToLocalize.entrySet()) {
+      final File localizedFile;
+      LocalizeResource resourceToLocalize = entry.getValue();
```

Comment:
This naming is very confusing. It has a `resourcesToLocalize` as a Map outside the loop. In fact, this variable only used once, so better chain the call:

```java
URI uri = entry.getValue().getURI();
```